doc/protocol.md

```diff
@@ -129,6 +129,38 @@ document:
 
 ## Specification
 
+### Introduction and version negotiation
+
+Immediately after establishing a connection, the client side must send an introduction message
+identifying versions of the protocol that it is able to use. The server responds with one of
+those versions, or an error indicating that no compatible version exists.
+
+This step exists to enable smoother protocol changes in the future, and for better compatibility
+with old clients.
+
+The client begins the connection by sending the following raw sequence of bytes:
+
+```
+0x49
+0x4D
+nVersions          // One byte, number of supported protocol versions
```

Comment:
This is identical to the 1.0 protocol's version negotiation; being compatible with that is useful.

It's one byte because that keeps byte-fiddling to a minimum (no endianness, etc). Not using a minimum means we can remove support for a version and have this recognized to display a useful message.